### PR TITLE
Porta Sleeper

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

Essentially, this copies most if not all of the sleeper code and cuts out the middle man of the console.

Things to do.

- [ ] Make it actually fold into a backpack. (have to figure out the clickanddrag proc)
- [ ] New sprites (give it wheels or something)
- [ ] Balance the cold cryostasis feature (it gets cold enough for Peri and QC to work their magic, Peri and QC are available as an injectable chem, It's so cold that Kelo and Tri together can't out heal cold burn damage, could up the temp to be less punishing but still effective.) 
- [ ] Make it able to inject up to 30u of each chem? (right now it's max 20, and you can't go over.)
- [ ] Remove all the bits that aren't actually necessary.
- [ ] Add it into the game. (req only? Medic points? SL points? one in Medbay storage? Researcher creation?)
- [ ] Give it an internal battery? (something that can be recharged/replaced. making it more portable and not as reliant upon power.)

**Any other suggestions are also welcome.**

## Why It's Good For The Game

This would make a fob/push base better able to treat the mildly wounded, and would provide another incentive to get power set up. It can be used by anyone, no medical skill (other than player) required.

## Changelog
:cl: Hughgent
add: New portable sleeper. With most of the capabilities of medbays sleeper, in a convenient backpack sized form factor.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Instead of copypasting the code, try putting it into procs that you are able to invoke from both the portable and the static sleepers.
This could ideally work as a component, but that is not trivial.
Poke me on discord when you have time and we can start trying to cook something.

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -479,3 +479,349 @@
...
 	add_fingerprint(usr)
+
+//PORTA-SLEEPER
+//Intended to be backpack sized when folded, and needs setup.
+
+/obj/machinery/sleeper/portable_sleeper
+	name = "Portable Sleeper"
+	desc = "A fancy folding bed with built-in injectors and a limited health scanner."
+	anchored = 0
+	available_chemicals = list("peridaxon" = "Peridaxon", "quickclot" = "Quick-Clot", "inaprovaline" = "Inaprovaline", "sleeptoxin" = "Soporific", "paracetamol" = "Paracetamol", "bicaridine" = "Bicaridine", "kelotane" = "Kelotane", "dylovene" = "Dylovene", "dexalin" = "Dexalin", "tricordrazine" = "Tricordrazine", "spaceacillin" = "Spaceacillin")
```
[the remark this answers is not in the document]
Comment (Hughgent) [reply code/game/machinery/Sleeper.dm]: I was gonna make it have a battery, as it uses (at the moment) 100 power per button press.

[the remark this answers is not in the document]
Comment (psykzz) [reply code/game/machinery/Sleeper.dm]: Battery does work, but depending on the map they can be easy to find.
Prefer requiring the area to be powered, if possible or limit the chemicals to require restocking, as it gives a reason to move about.

File: code/game/machinery/Sleeper.dm [hunk 2]
```diff
@@ -479,3 +479,349 @@
...
+	if(ismob(user.pulledby))
+		var/mob/grabmob = user.pulledby
+		grabmob.stop_pulling()
+	user.stop_pulling()
+	if(!user.forceMove(src))
+		return
+	visible_message("[user] climbs into the sleeper.", 3)
+	update_use_power(2)
+	occupant = usr
+	start_processing()
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/game/machinery/Sleeper.dm]: Start_processing is currently the preferred way for machines

File: code/game/machinery/Sleeper.dm [hunk 3]
```diff
@@ -479,3 +479,349 @@
...
+/obj/machinery/sleeper/portable_sleeper/Initialize()
+	. = ..()
+	if(orient == "RIGHT")
+		icon_state = "sleeper_0-r"
+		
+/obj/machinery/sleeper/portable_sleeper/Destroy()
+	occupant?.in_stasis = FALSE //clean up; end stasis; remove from processing
+	occupant = null
+	STOP_PROCESSING(SSobj, src)
+	stop_processing()
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/game/machinery/Sleeper.dm]: Stop_processing is currently the preferred way for machines

**Pull request closed without merging**